client/src/components/InputField.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <md-field :class="fieldClass">
+    <div v-if="inputType == 'range'">
+      <label>{{label}}</label>
+      <span class="md-helper-text">{{value}}</span>
+    </div>
+    <div v-if="inputType == 'checkbox'">
+    </div>
+    <div v-else>
+      <label>{{label}}</label>
+    </div>
+
+    <div v-if="inputType == 'checkbox'">
+      <md-checkbox
+      :value="inputStr"
+      :type="inputType"
+      v-on:input="handleInput"
+      v-model="value"
+      :id="inputName"
+      />{{label}}
+    </div>
+    <div v-else-if="inputType == 'option'">
+      <md-select
+      :value="inputStr"
+      v-on:input="handleInput"
+      :type="inputType"
+      v-model="value"
+      :id="inputName"
+      md-dense
+      >
+        <md-option
+          v-for='(name, value) in options'
+          :key='value'
+          :value='value'
+          >{{ name }}</md-option
+        >
+      </md-select>
+    </div>
+    <div v-else>
+      <md-input
+      :value="inputStr"
+      v-on:input="handleInput"
+      :type="inputType"
+      :step="step"
+      :min="min"
+      :max="max"
+      :class="fieldClass"
+      v-model="value"
+      />
+    </div>
+    <span class="md-error">{{ errorStr }}</span>
+  </md-field>
+</template>
+
+<script>
+export default {
+  name: 'InputField',
+  props: {
+    step: Number,
+    min: Number,
+    max: Number,
+    label: String,
+    inputName: String,
+    inputType: String,
+    options: Object,
+    currentValue: [Number, String],
+    setCurrentValue: Function,
+  },
+  data: () => ({
+    inputIsInvalid: false,
+    inputStr: null,
+    errorStr: null,
+    value: null,
+  }),
+  computed: {
+    fieldClass() {
+      return {
+        'md-invalid': this.inputIsInvalid,
+      };
+    },
+  },
+  watch: {
+    currentValue() {
+      if (this.currentValue !== this.inputStr) {
+        this.inputStr = this.currentValue;
+      }
+    },
+  },
+  methods: {
+    handleInput(value) {
+      if (typeof value === 'string' || value instanceof String) {
+        this.inputIsInvalid = false;
+        this.setCurrentValue(value);
+      } else if (!Number.isNaN(value)) {
```

Comment:
At the moment when values are entered into the `Input` component they don't seem to show an error when they aren't correct. See below for a screenshot:

![Screenshot](https://i.imgur.com/53uN151.png)

This might be because of this section of code. Right now the type of the value coming from the inputs will always be a string, so it will never error.

Reply:
Good to know I didn't realize they all returned strings.

Reply:
![image](https://user-images.githubusercontent.com/46701725/97828605-d03f4400-1c84-11eb-8c2d-16ee26afefb7.png)
fixed